AppController/lib/taskqueue.rb

```diff
@@ -168,21 +167,24 @@ def self.start_slave(master_ip, clear_data, verbose)
     Djinn.log_debug('Waiting for RabbitMQ on master node to come up')
     HelperFunctions.sleep_until_port_is_open(master_ip, SERVER_PORT)
 
-    # we now reset it to join the head node. To do this we need the
-    # hostname of the master node. We go through few options:
-    # - the old one is to look into /etc/hosts for it
-    # - another one is to just try to resolve it
-    # - finally we give up and use the IP address
-    master_tq_host = `cat /etc/hosts | grep #{master_ip} | tr -s \" \" | cut -d \" \" -f2`.chomp
-    if master_tq_host.empty?
-      begin
-        master_tq_host = Resolv.getname(master_ip)
-      rescue Resolv::ResolvError
-        # We couldn't get the name: let's try to use the IP address.
-        master_tq_host = master_ip
+    # Look up the TaskQueue master's hostname (not the fqdn). To resolve
+    # it we use Addrinfo.getnameinfo since Resolv will not use /etc/hosts
+    # and this could cause issues on private clusters.
+    master_tq_host = nil
+    begin
+      master_tq_host = Addrinfo.ip(master_ip).getnameinfo[0]
+      unless (master_tq_host =~ /^[[:digit:]]/).nil?
```

Comment:
`unless/else` is like a double negative, which makes it more difficult to read. In this case, I think
```
if master_tq_host =~ /^[[:digit:]]/
  Djinn.log_warn...
else
  master_tq_host =...
end
```
would be clearer.